CleverCloud/clever-components
# Enable typechecking & `ts-lit-plugin` for IDEs / Editors

Fixes #912 
Fixes #916

## What does this PR do?

- Fixes the `tsconfig.json` file with `allowJs: true`. Thanks to this, IDEs may rely on `tsserver` or a TypeScript LSP to benefit from autoimport and other advanced features,
- Enables typechecking for the whole project in our IDEs (not in the CI),
- Adds the `ts-lit-plugin` to our dev dependencies so we can benefit from hints in our IDEs.

## How to review?

- Check the commits,
- Check out locally, **run `npm install`**  and open some component files, you should be able to:
  -  see type errors, missing type imports, etc.,
  - with VSCode, go to `cc-badge` constructor and type `Resize`, you should get suggestions for `ResizeController`. If you accept the suggestion, it should auto-import the `ResizeController`.
  - in the `cc-badge` component, comment the `cc-icon` import => you should get an error on the `<cc-icon ...>` tag in the render. You also should have a quick fix action to add the import (not great because it adds `"` and skips the file extension even when the language service is properly set :shrug: )

There are other things this PR enables but it's enough to be sure everything is working.
 
As long as the CI does not break, it means this PR does not mess with the rest of our tooling so we should be good to go.


**Note**: the `go to def` on a custom element does not seem to work on WebStorm even with `ts-lit-plugin` :disappointed:. It works with other editors. 

## TODO Flo

- [x] copy/paste options from `tsconfig.json` to the config used for the typechecking once we're settled on the options we want to use.
